File: examples/ApiTest.cs
using System;
using Roblox.Instances;
using Roblox.Services;
using Roblox.Datatypes;

public class ApiTest
{
    public async System.Threading.Tasks.Task RunTest()
    {
        // Test GetService
        var players = Roblox.Game.GetService<Players>();
        
        // Test Event
        players.PlayerAdded += async (player) => {
            Console.WriteLine("Player joined!");
            await DoSomethingAsync();
        };

        // Test Method Call (Colon)
        var allPlayers = players.GetPlayers();

        // Test Property Access (Dot)
        Console.WriteLine("Max players: " + players.MaxPlayers);

        // Test Instance Creation
        var part = new Part();
        part.Name = "TestPart";
        part.Position = new Vector3(0, 10, 0);
        part.Parent = Roblox.Game.Workspace;

        // Test Null Conditional
        var name = part?.Name;
        var nested = part?.Parent?.Name;
        Console.WriteLine("Part name: " + name);

        // Test Async/Await (Stripping)
        await DoSomethingAsync();
    }

    private async System.Threading.Tasks.Task DoSomethingAsync()
    {
        Console.WriteLine("Doing something async...");
    }
}

File: examples/ApiTest.lua
local ApiTest = {}
ApiTest.__index = ApiTest
function ApiTest.new()
    local self = setmetatable({}, ApiTest)
    return self
end
function ApiTest:RunTest()
    local players = game:GetService("Players")
    players.PlayerAdded:Connect(function(player)
        task.spawn(function()
            print("Player joined!")
            self:DoSomethingAsync()
        end)
    end)
    local allPlayers = players:GetPlayers()
    print("Max players: " .. players.MaxPlayers)
    local part = Instance.new("Part")
    part.Name = "TestPart"
    part.Position = Vector3.new(0, 10, 0)
    part.Parent = workspace
    local name = part and part.Name
    local nested = part and part.Parent and part.Parent.Name
    print("Part name: " .. name)
    self:DoSomethingAsync()
end
function ApiTest:DoSomethingAsync()
    print("Doing something async...")
end
return ApiTest
